spec/features/admin/bulk_product_update_spec.rb

```diff
@@ -516,37 +516,61 @@
+      it "shows an edit button for products, which takes the user to the standard edit page for that product" do
         expect(page).to have_selector "a.edit-product", count: 2
 
-        new_window = window_opened_by do
-          within "tr#p_#{p1.id}" do
-            find("a.edit-product").click
-          end
+        within "tr#p_#{p1.id}" do
+          find("a.edit-product").click
         end
 
-        within_window new_window do
-          expect(URI.parse(current_url).path).to eq "/admin/products/#{p1.permalink}/edit"
-          page.execute_script('window.close()')
+        expect(URI.parse(current_url).path).to eq "/admin/products/#{p1.permalink}/edit"
+      end
+
+      it "shows an edit button for products, which takes the user to the standard edit page for that product, url includes selected filter" do
+        expect(page).to have_selector "a.edit-product", count: 2
+
+        # Set a filter
+        select2_select p1.supplier.name, from: "producer_filter"
+        apply_filters
+
+        within "tr#p_#{p1.id}" do
+          find("a.edit-product").click
         end
+
+        uri = URI.parse(current_url)
+        expect("#{uri.path}?#{uri.query}").to eq "/admin/products/#{p1.permalink}/edit?producerFilter=#{p1.supplier.id}"
       end
 
-      it "shows an edit button for variants, which takes the user to the standard edit page for that variant in a new window" do
+      it "shows an edit button for variants, which takes the user to the standard edit page for that variant" do
         expect(page).to have_selector "a.view-variants"
         all("a.view-variants").each(&:click)
 
         expect(page).to have_selector "a.edit-variant", count: 2
 
-        new_window = window_opened_by do
-          within "tr#v_#{v1.id}" do
-            find("a.edit-variant").click
-          end
+        within "tr#v_#{v1.id}" do
+          find("a.edit-variant").click
         end
 
-        within_window new_window do
-          expect(URI.parse(current_url).path).to eq "/admin/products/#{v1.product.permalink}/variants/#{v1.id}/edit"
-          page.execute_script('window.close()')
+        uri = URI.parse(current_url)
+        expect(URI.parse(current_url).path).to eq "/admin/products/#{v1.product.permalink}/variants/#{v1.id}/edit"
+      end
+
+      it "shows an edit button for variants, which takes the user to the standard edit page for that variant, url includes selected filter" do
+        expect(page).to have_selector "a.view-variants"
+        all("a.view-variants").each(&:click)
+
+        expect(page).to have_selector "a.edit-variant", count: 2
+
+        # Set a filter
+        select2_select p1.supplier.name, from: "producer_filter"
+        apply_filters
+
+        within "tr#v_#{v1.id}" do
+          find("a.edit-variant").click
         end
+
+        uri = URI.parse(current_url)
+        expect("#{uri.path}?#{uri.query}").to eq "/admin/products/#{v1.product.permalink}/variants/#{v1.id}/edit?producerFilter=#{p1.supplier.id}"
```

Comment:
same here, I think this is
edit_admin_product_variant_path(v1.product.permalink, v1.id, producerFilter: p1.supplier.id)

Reply:
fixed